### ISSUE TITLE
Vacuum doesn not connect to MQTT broker

### ISSUE THREAD

**Issue by vdg1** [rand256/valetudo#231]
**Describe the bug**
I have installed Valetudo RE 0.9.4 and MQTT is configured in config.json. I have tested the MQTT broker using Home Assistant and MQTT Explorer. All that is working fine - but the vacuum does not connect to the MQTT broker. Looking at the source of MQTTclient.js the robot should log message like "Connected to MQTT.." or similar. I don't see these messages in the valetudo.log.

My config.json:

  "mqtt": {
    "enable": true,
    "identifier": "Erna",
    "topicPrefix": "valetudo",
    "autoconfPrefix": "homeassistant",
    "broker_url": "mqtt://xx:yy@192.168.1.zzz:1883",
    "provideMapData": true,
    "mapSettings": {
      "drawPath": true,
      "drawCharger": true,
      "drawRobot": true,
      "border": 2,
      "scale": 4
    },
    "caPath": "",
    "qos": 0
  },

Valetudo.log:
root@rockrobo:~# tail -f /var/log/upstart/valetudo.log
Waiting for 30 sec after boot... done.
2020-05-07T20:36:42.522Z Loading configuration file: /mnt/data/valetudo/config.j                                       son
2020-05-07T20:36:42.692Z Dummycloud is spoofing 203.0.113.1:8053 on 127.0.0.1:80                                       53
2020-05-07T20:36:42.695Z Webserver running on port 80
2020-05-07T20:36:58.993Z Failed to get handshake for message: get_status [] { re                                       tries: 0, retriesHS: 100 }
2020-05-07T20:37:02.618Z Failed to get handshake for message: get_timezone [] {                                        retries: 0, retriesHS: 100 }
2020-05-07T20:37:02.738Z Failed to get handshake for message: get_status [] { re                                       tries: 0, retriesHS: 100 }
2020-05-07T20:37:05.613Z Failed to get handshake for message: get_status [] { re                                       tries: 0, retriesHS: 100 }
2020-05-07T20:37:08.031Z Failed to get handshake for message: get_status [] { re                                       tries: 0, retriesHS: 100 }
2020-05-07T20:37:10.471Z Failed to get handshake for message: get_status [] { re                                       tries: 0, retriesHS: 100 }
2020-05-07T20:37:22.948Z Robot (re)connected
2020-05-07T20:37:23.068Z Probed last id = 1001 using get_status (82 retries)

**How to Reproduce**

**Vacuum Model:**  Gen2/S5

**Valetudo Version:** 0.9.4

Any ideas what could be wrong? I'd appreciate any information. Thanks!

**Comment by pidator** [comment]
You've rebooted the robot after changing the config.json file?

**Comment by vdg1** [comment]
Yep, multiple times

**Comment by vdg1** [comment]
I tried MQTT with and without authentication, to no avail.

**Comment by rand256** [comment]
Which firmware version do you have? How exactly did you installed valetudo? Theoretically you should never see "Failed to get handshake for message" errors in the log though not sure how could that be connected with the issue.

**Comment by vdg1** [comment]
This is from the web interface:
![image](https://user-images.githubusercontent.com/2090293/81398553-b388e000-9129-11ea-8992-bc37198d6090.png)


Sounds like you suggest I factory reset the bot and reinstall Valetudo RE?

**Comment by rand256** [comment]
No, I don't suggest that. I asked about how you build the image cause I'm interested why do you have "Failed to get handshake" error. Though you still may want to try reflashing any vacuumz-based image, while factory reset most likely would be irrelevant.

**Comment by vdg1** [comment]
Sorry, misunderstood that. I used dustbuilder to build Valetudo 0.9.2 and I tried that. Didn't get MQTT to work, so I downloaded the deb package of Valetudo 0.9.4 and installed that with dpkg. With both versions I saw the handshake errors in the log.

**Comment by vdg1** [comment]
After some trial and error I accidentally had a typo in the config.json ("trrue" instead of "true"). That caused a reset of the config.json when I restarted the valetudo service. All zones and go to point definitions were gone, but never mind... because Valetudo then connected to the MQTT broker after I reconfigured the MQTT settings in the config.json file. So some configuration fault in the config.json made Valetudo not to connect to the MQTT broker in the first place. Just wonder what I did wrong in the config.json.....
Anyway. case closed
